src/Auditable.php

```diff
@@ -688,18 +691,20 @@ public function auditAttach(string $relationName, $id, array $attributes = [], $
      * @param mixed $ids
      * @param bool $touch
      * @param array $columns
+     * @param BelongsToMany|null $relationship
      * @return int
      * @throws AuditingException
      */
-    public function auditDetach(string $relationName, $ids = null, $touch = true, $columns = ['*'])
+    public function auditDetach(string $relationName, $ids = null, $touch = true, $columns = ['*'], $relationship = null)
     {
         if (!method_exists($this, $relationName) || !method_exists($this->{$relationName}(), 'detach')) {
```

Comment:
I think it would be better to add a closure. this change make `$relationName` useless

Reply:
I think the relationship name is necessary because it is used in the dispatchRelationAuditEvent method. Do you have any suggestions on how to get the relationship name another way while maintaining the functionality I'm trying to add?

However, the flaws you pointed out really exist.

Reply:
>Do you have any suggestions on how to get the relationship name another way while maintaining the functionality I'm trying to add?

>I think it would be better to add a closure

Already answered, a closure, Laravel uses closures for these cases, that will maintain consistency